### PR TITLE
ref(sentry apps): Improve Sentry App Component endpoint errors

### PR DESCRIPTION
**Pull request by Christinarlong**
tl;dr Instead of returning a json with `error: True` return a json with `error: "ValidationError: bruh"`

Expose sentry app component endpoint errors. Instead of returning a boolean indicating that an error occurred while trying to make the [component](https://docs.sentry.io/organization/integrations/integration-platform/ui-components/) we return the error message.

Context for SentryAppComponents: Some [SentryAppComponents](https://docs.sentry.io/organization/integrations/integration-platform/ui-components/) will [reach out to a specified URI to get dropdown options](https://docs.sentry.io/organization/integrations/integration-platform/ui-components/#uri-guidelines) from. If the request to the URI fails, currently we catch the failure, raise an exception and in the S[entryAppComponentEndpoint and append the `component_uuid` to an errors list](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_apps/api/endpoints/sentry_app_components.py#L84). Then in the [serializer](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_apps/api/serializers/sentry_app_component.py#L17) we check the errors list to see if the serialized component's uuid is present and if so we set an error flag on the component to True. The frontend then reads the returned json and if the error flag is True, we disable the flagged component (grey out). 

Tested all the errors & messages, but this can go in after broader changes to errors for SentryApps outlined [here](https://www.notion.so/sentry/Sentry-App-Errors-1508b10e4b5d80babd74f7f0a3eada6c?pvs=4)

UPDATE: Relies on https://github.com/getsentry/sentry/pull/83678

### CODE REVIEW

File: src/sentry/sentry_apps/api/serializers/sentry_app_component.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def serialize(self, obj, attrs, user, **kwargs):
             "uuid": str(obj.uuid),
             "type": obj.type,
             "schema": obj.schema,
-            "error": True if str(obj.uuid) in errors else False,
+            "error": errors.get(str(obj.uuid), ""),
```
Comment (Christinarlong) [inline src/sentry/sentry_apps/api/serializers/sentry_app_component.py:17]: In the [above](https://github.com/getsentry/sentry/pull/81167/files#diff-00e9151c57a98b8e0d198312c2dcdb6eca386cc3936f121345e427deb738b8a7R85-R95) we catch all errors (assuming they're raised), so there should never be the case where a component in in `errors` but the message is an empty string.

Comment (cathteng) [reply src/sentry/sentry_apps/api/serializers/sentry_app_component.py:17]: what does it look like on the frontend when we return the entire error instead of a bool?

Comment (Christinarlong) [reply src/sentry/sentry_apps/api/serializers/sentry_app_component.py:17]: Nothing will change for frontend appearance wise. Currently we use that boolean to determine whether or not we [need to gray out the UI component](https://github.com/getsentry/sentry/blob/master/static/app/components/sentryAppComponentIcon.tsx#L39-L41) and display a generic 'couldn't connect to [SentryApp]' message. That message and appearance will stay the same, the new errors will be exposed via the network requests.

Comment (cathteng) [reply src/sentry/sentry_apps/api/serializers/sentry_app_component.py:17]: should that be a change we make -- to surface the more explicit error message?

**Review by cathteng — APPROVED** [review]
do we need to update API docs for this change?

File: src/sentry/sentry_apps/api/serializers/sentry_app_component.py (hunk 1 above)
Comment (Christinarlong) [reply src/sentry/sentry_apps/api/serializers/sentry_app_component.py:17]: we could, but that would be a future PR. Our integrators are all developers who will check the network reqs to debug cust issues, so this change is needed either way to provide that visibility.

**Comment by Christinarlong** [comment]
> do we need to update API docs for this change?

ney this is a private endpoint
